From 541725e743289af798091198a98eceb72d8eeaf4 Mon Sep 17 00:00:00 2001
From: Rob Latham <robl@mcs.anl.gov>
Date: Tue, 19 Dec 2017 15:49:13 -0600
Subject: [PATCH] my c++ installation needed an explicit cast

---
 examples/00_init/server.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/00_init/server.cpp b/examples/00_init/server.cpp
index 8bea7359..6128a9b6 100644
--- a/examples/00_init/server.cpp
+++ b/examples/00_init/server.cpp
@@ -6,7 +6,7 @@ namespace tl = thallium;
 int main(int argc, char** argv) {
 
     tl::engine myEngine("bmi+tcp://127.0.0.1:1234", THALLIUM_SERVER_MODE);
-    std::cout << "Server running at address " << myEngine.self() << std::endl;
+    std::cout << "Server running at address " << std::string(myEngine.self()) << std::endl;
 
     return 0;
 }
-- 
2.14.1

