From 4a1483cf04b5d4eefd2fa6387c922b2edc76695b Mon Sep 17 00:00:00 2001
From: Rob Latham <robl@mcs.anl.gov>
Date: Thu, 7 Sep 2017 15:00:08 -0500
Subject: [PATCH] Werror is a little aggressive

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index a4a22fad..69a15179 100644
--- a/configure.ac
+++ b/configure.ac
@@ -117,10 +117,10 @@ AS_IF([test "$PE_ENV" = "PGI"],
       [cflags="-fast -Msmartalloc -Mnobuiltin $DEBUG_FLAGS"])
 AS_IF([test "$PE_ENV" = "GNU"],
       AC_SUBST([CC],["gcc"])
-      [cflags="-O3 -pthread -fno-builtin -Wall -Werror $DEBUG_FLAGS"])
+      [cflags="-O3 -pthread -fno-builtin -Wall $DEBUG_FLAGS"])
 AS_IF([test "$PE_ENV" = "INTEL"],
       AC_SUBST([CC],["icc"])
-      [cflags="-O3 -pthread -fno-builtin -Wall -Werror $DEBUG_FLAGS"])
+      [cflags="-O3 -pthread -fno-builtin -Wall $DEBUG_FLAGS"])
 
 # Cray compiler is not ready yet; needs libtool support
 #AS_IF([test "$PE_ENV" = "CRAY"],
-- 
2.14.1

